swiftsimio/accelerated.py

```diff
@@ -168,3 +180,148 @@ def index_dataset(handle: Dataset, mask_array: np.array) -> np.array:
     ranges = ranges_from_array(mask_array)
 
     return read_ranges_from_file(handle, ranges, output_size, output_type)
+
+
+################################ ALEXEI: playing around with better read_ranges_from_file implementation #################################
+
+def get_chunk_ranges(ranges, chunk_size, array_length):
+    chunk_ranges = []
+    for bound in ranges:
+        lower = int(np.floor(bound[0]/chunk_size))*chunk_size
+        upper = min(int(np.ceil(bound[1]/chunk_size))*chunk_size, array_length)
+
+        if len(chunk_ranges) > 0:
+            assert(lower >= chunk_ranges[-1][0])
+            assert(upper >= chunk_ranges[-1][1])
+            if lower > chunk_ranges[-1][1]:
+                chunk_ranges.append([lower, upper])
```

Comment:
I don't really understand what's going on here - can you provide some comments?

Reply:
This is just to check whether the chunk we want to add to the list overlaps with the last one (to prevent e.g. [[1,10],[10, 20]] or [[1,30], [20,30]]). I think most commonly it would happen if you have two ranges within one chunk, you want to prevent adding duplicates. I'll add comments to this effect in the next commit along with the other small changes.